config/regex.js

```diff
@@ -59,7 +59,7 @@ module.exports = {
     ENV_NAME: /^[A-Z_][A-Z0-9_]*$/,
     // Repo checkout url. For example: https://github.com/screwdriver-cd/data-schema.git#branchName or git@github.com:screwdriver-cd/data-schema.git
     // eslint-disable-next-line max-len
-    CHECKOUT_URL: /^(?:(?:https?|git):\/\/)?(?:[^@]+@)?([^/:]+)(?:\/|:)([^/]+)\/(.+?)(?:\.git)?(#.+)?$/,
+    CHECKOUT_URL: /^(?:(?:https:\/\/(?:[^@/:\s]+@)?)|git@)+([^/:\s]+)(?:\/|:)([^/:\s]+)\/([^\s]+?)(?:\.git)?(#[^\s]+)?$/,
```

Comment:
Seems like this regex captures input such as, `https://fjahja@github.com:org/repo.git` and treats `https://fjahja@` as a single group.

I think we can simplify the first part of the regex from `(?:(?:https:\/\/(?:[^@/:\s]+@)?)|git@)+` to 
`(?:https:\/\/|[^@/:\s]+@)`.

Reply:
That regex won't work for bitbucket checkoutUrls under a team/org like `https://tkyi@bitbucket.org/d2lam2/test.git#branch`